lib/Magic.php

```diff
@@ -11,26 +11,38 @@ class Magic
     public $token;
     public $api_secret_key;
     public $user;
-    public $wallet_type;
+    public $client_id;
 
     public function __construct(
         $api_secret_key = null,
         $timeout = TIMEOUT,
         $retries = RETRIES,
-        $backoff_factor = BACKOFF_FACTOR
+        $backoff_factor = BACKOFF_FACTOR,
+        $client_id = null
     ) {
         $this->api_secret_key = $api_secret_key;
-        $this->token = new \MagicAdmin\Resource\Token();
+        $request_client = new \MagicAdmin\HttpClient($api_secret_key, $timeout, $retries, $backoff_factor);
+        if ($client_id != null) {
+          $this->client_id = $client_id;
+        } else {
+          $this->client_id = $this->_get_client_id($request_client);
+        }
+        $this->token = new \MagicAdmin\Resource\Token($this->client_id);
```

Comment:
Missed this in the python PR but what's the reason allowing a user to enter a client ID? Are there negative consequences trusting the value they pass in or is there validation we do down then line

Reply:
This is for clients who don't want to make network calls and would prefer to pass in the client ID themselves. Ideally our secret keys would encode our client IDs :(